proposals/yaml_signing_and_verification.md

```diff
@@ -0,0 +1,168 @@
+kind: ClusterPolicy
+metadata:
+  name: check-image
+spec:
+  rules:
+    - name: check-manifest
+      match:
+        resources:
+          kinds:
+            - ClusterPolicy
+            - Policy
+      validate:
+        manifests:
+          - key: |-
+            -----BEGIN PUBLIC KEY-----
+                      ...
+            -----END PUBLIC KEY-----
+```
+
+This option of signing and verifying YAMLs is compatible with GitOps workflows and does not require an OCI registry to store signed resource definitions.
+
+## Implementation
+
+Although referred to as “YAML signing” this feature may be more accurately described as verifying the integrity of a Kubernetes manifest or resource declaration. YAML is used as a canonical format during signing and verification.
+
+Since Kubernetes resources are typically mutated during admission controls by native and 3rd party controllers, this feature needs to be able to allow certain mutations, and ignore them during serialization and verification.
+
+## Requirements
+
+1. Kyverno should support a new `validate.manifests` declaration to allow verifying a YAML has been signed by one or more signatures.
+2. The declaration must support a list of keys and allow logical AND / OR operations to allow signing by at least one of the keys for key rotation, a minimum count of keys, and multiple keys e.g. team and group level.  
+3. A Kyverno policy rule for YAML verification should automatically allow (i.e. ignore) mutations performed by “trusted” controllers such as Kubernetes controllers and  Kyverno mutate rules.  
+
+
+## Drawbacks
+
+* Each resource needs to be signed vs. creating a resource bundle for a set of resources.
+* Calculating the canonical form of the resource requires a “dry-run” to eliminate changes made based on defaults and other admission controllers.
+* Performing a dry-run create requires Kyverno to have permissions to create the resource.
+
+## Known Implementations
+
+The Stolstron project Contains OpenShift/OKD compatibility projects for Open Cluster Management. Stolostron Integrity Shield provides:
+
+* Signing and verification with GPG, X.509 certificates, and Sigstore cosign
+* An admission controller with custom policy resources [ManifestIntegrityConstraint ](https://github.com/stolostron/integrity-shield/blob/master/docs/README_GETTING-STARTED-TUTORIAL.md#create-a-sample-manifestintegrityconstraint)and [ManifestIntegrityState](https://github.com/stolostron/integrity-shield/blob/master/docs/README_GETTING-STARTED-TUTORIAL.md#check-the-resource-integrity-status-from-the-manifestintegritystate-generated-by-observer).
+* An Gatekeeper/OPA extension
+* An operator
+
+More information at: [https://github.com/stolostron/integrity-shield#integrity-shield](https://github.com/stolostron/integrity-shield#integrity-shield) 
+
+
+# Alternatives
+
+## Signed OCI Bundle
+
+Kyverno policies can be stored in an OCI registry and signed as an OCI bundle. This is discussed in:  
```

Comment:
Discussion link is absent.

Reply:
It would be nice to see pros and cons for the OCI registry approach laid out here so we can compare and contrast, especially since that isn't your recommendation.